src/Microsoft.Azure.ServiceBus/SubscriptionClient.cs

```diff
@@ -444,15 +448,28 @@ public Task AddRuleAsync(string ruleName, Filter filter)
             description.ValidateDescriptionName();
             MessagingEventSource.Log.AddRuleStart(this.ClientId, description.Name);
 
+            bool isDiagnosticsEnabled = ServiceBusDiagnosticSource.IsEnabled();
+            Activity activity = isDiagnosticsEnabled ? this.diagnosticSource.AddRuleStart(description) : null;
+            Task addRuleTask = null;
+
             try
             {
-                await this.InnerSubscriptionClient.OnAddRuleAsync(description).ConfigureAwait(false);
+                addRuleTask = this.InnerSubscriptionClient.OnAddRuleAsync(description);
+                await addRuleTask.ConfigureAwait(false);
             }
             catch (Exception exception)
             {
                 MessagingEventSource.Log.AddRuleException(this.ClientId, exception);
+                if (isDiagnosticsEnabled)
+                {
+                    this.diagnosticSource.ReportException(exception);
+                }
                 throw;
             }
+            finally
+            {
+                this.diagnosticSource.AddRuleStop(activity, description, addRuleTask?.Status);
```

Comment:
>this.diagnosticSource.AddRuleStop(activity, description, addRuleTask?.Status); [](start = 16, length = 78)

wrap inside isDiagnosticsEnabled?

Reply:
Similarly at other places

---
In reply to: [149190095](https://github.com/Azure/azure-service-bus-dotnet/pull/370#discussion_r149190095) [](ancestors = 149190095)

Reply:
if IsDiagnosticsEnabled is false, activity is null, all Stop methods do nothing if activity is null